superset-frontend/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -145,9 +148,18 @@ const ExploreChartPanel = ({
     getItem(LocalStorageKeys.chart_split_sizes, INITIAL_SIZES),
   );
 
+  const [showDatasetModal, setShowDatasetModal] = useState(false);
+  console.log(showDatasetModal);
+
+  const metaDataRegistry = getChartMetadataRegistry();
+  const { useLegacyApi } = metaDataRegistry.get(vizType);
+  const vizTypeNeedsDataset = useLegacyApi && datasource.type === 'query';
```

Comment:
I know duck typing isn't going to be easy here, but for readability and extendability, I think checking that it's not a dataset would make more sense rather than checking that it's a query.. and then later query or savedquery, etc.

Reply:
I think that makes a lot of sense. Changed.